fix(KFLUXBUGS-1680):  filter records on creationTimestamp

caugello opened the pull request:
## Fixes 
[KFLUXBUGS-1680](https://issues.redhat.com//browse/KFLUXBUGS-1680)


## Description
Querying pipelineruns or records may include all component names making the query to long and throws an HTTP error 414 or 431. Component names were included in the filter to avoid a visual issue when recreating an application with the same name, previously ran pipelineruns are displayed.

The component filter array is removed and uses the record creationTimestamp filter to be greater than the application creationTimestamp.


## Type of change
<!-- Please delete options that are not relevant. -->

- [ ] Feature
- [X] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 
<!-- If change affects UI in any way, tag relevant UX people and add screenshots/gifs  -->


## How to test or reproduce?
Create a large number of components with large names.

[ ] Test A 
Initial issue:
When the query is too long, we should hit an http error 414 or as I do, a 431 locally.

 [ ] Test B 
Once change applied,  the filter includes a creationTimestamp attribute and removed the component matchExpression filter.

**Test Configuration(s)**:
I have an application with 240 components, named `devfile-sample-other-very-long-meaningless-name-${n}`

## Browser conformance: 
<!-- To mark tested browsers, use [x] -->
- [ ] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge


<!-- ## Checklist: -->

<!-- 
- [ ] Code follows the style guidelines
- [ ] Self-reviewed the code
- [ ] Added comments in hard-to-understand areas
- [ ] Made corresponding changes to the documentation
- [ ] Changes generate no new warnings
- [ ] Added tests that prove this fix is effective or that the feature works
- [ ] New and existing unit tests pass locally with new changes
- [ ] Any dependent changes have been merged and published in downstream modules 
-->

caugello commented:
@sahil143 PTAL if this would be a suitable approach.

sahil143 commented:
> @sahil143 PTAL if this would be a suitable approach.

@caugello We added all components to the api query because earlier if user deleted an application, pipeline runs in that application were not removed so in some cases user created a new application with same name and they were seeing pipeline runs from deleted application. We need to check is this is still the case if It's not, we can stop sending component names in the API request otherwise we need to raise this with the backend team. Ideally we shouldn't need to send all of the component names in the api request.

caugello commented:
> > @sahil143 PTAL if this would be a suitable approach.
> 
> @caugello We added all components to the api query because earlier if user deleted an application, pipeline runs from that application were not removed so in some cases user created a new application with same name and they were seeing pipeline runs from deleted application. We need to check is this is still the case if It's not, we can stop sending component names in the API request otherwise we need to raise this with the backend team. Ideally we shouldn't need to send all of the component names in the api request, only application name should suffice.

This is still the case. Deleted an application, re-created it. Pipeline-runs history from previous application with same name still present

caugello commented:
Applied changes to filter on creationTimestamp and removed component names filters causing the issue.

caugello commented on src/utils/tekton-results.ts:
```diff
@@ -167,8 +183,6 @@ export const selectorToFilter = (selector?: Selector) => {
       if (matchExpressions) {
         filter = AND(filter, expressionsToFilter(matchExpressions));
       }
-    } else {
-      filter = labelsToFilter(selector as MatchLabels);
```
This is overwriting the filter. If a selector expression doesn't include a matchlabel or matchexpressions it will always execute the else part which is setting the incorrect filter hence the change in test output as well.  `filterByName` should have a `startsWith` expression instead of a label key assertion.

CryptoRodeo commented:
Overall LGTM, but I'll wait for Sahil to chime in :smile:

sahil143 commented on src/hooks/useBuildPipelines.ts:
```diff
@@ -22,22 +22,12 @@ export const useBuildPipelines = (
             [PipelineRunLabel.APPLICATION]: applicationName,
             [PipelineRunLabel.PIPELINE_TYPE]: PipelineRunType.BUILD,
           },
-          ...(includeComponents &&
-            componentNames?.length > 0 && {
-              matchExpressions: [
-                {
-                  key: PipelineRunLabel.COMPONENT,
-                  operator: 'In',
-                  values: componentNames,
-                },
-              ],
-            }),
```
we should add the creation timestamp filter here also. This also has the potential of fetching the deleted pipeline runs

caugello replied:
updated all where requested

sahil143 commented on src/hooks/useLatestBuildPipelines.ts:
```diff
@@ -29,16 +29,9 @@ export const useLatestBuildPipelines = (
             [PipelineRunLabel.APPLICATION]: applicationName,
             [PipelineRunLabel.PIPELINE_TYPE]: PipelineRunType.BUILD,
           },
-          matchExpressions: [
-            {
-              key: PipelineRunLabel.COMPONENT,
-              operator: 'In',
-              values: neededNames,
-            },
-          ],
```
same as above

sahil143 commented on src/hooks/usePACStatesForComponents.ts:
```diff
@@ -83,16 +83,11 @@ const usePACStatesForComponents = (components: ComponentKind[]): PacStatesForCom
             [PipelineRunLabel.COMMIT_EVENT_TYPE_LABEL]: PipelineRunEventType.PUSH,
           },
           matchExpressions: [
-            {
-              key: PipelineRunLabel.COMPONENT,
-              operator: 'In',
-              values: neededNames,
-            },
```
same as above

sahil143 commented on src/hooks/usePipelineRuns.ts:
```diff
@@ -274,22 +274,12 @@ export const usePipelineRunsForCommit = (
           matchLabels: {
             [PipelineRunLabel.APPLICATION]: applicationName,
           },
-          ...(componentsLoaded &&
-            componentNames?.length > 0 && {
-              matchExpressions: [
-                {
-                  key: PipelineRunLabel.COMPONENT,
-                  operator: 'In',
-                  values: componentNames,
-                },
-              ],
-            }),
```
same as above

caugello commented:
/retest

sahil143 reviewed (approved):
/lgtm

caugello commented:
/retest
